### ISSUE TITLE
Adapt HTTP/2 client

### ISSUE THREAD

**Issue by minwoox** [line/centraldogma-go#19]
Golang supports HTTP/2 from version 1.6, I think it is pretty mature in Golang and we can take advantage of HTTP/2.
While I was figuring out how to adapt it in our client, I found two situations that we have to consider:
- The `oauth2` client we use internally does not support HTTP/2 yet.
- It seems like h2c is not supported in the client-side. See [this](https://github.com/golang/net/tree/master/http2/h2c/h2c.go).

For the first one, I think we can create the `oauth2` client directly something like:
```
// The code creating a client in oauth2
return &http.Client{
	Transport: &Transport{
		Base:   internal.ContextClient(ctx).Transport,
		Source: ReuseTokenSource(nil, src),
	},
}
// Create our own using the exported Struct and field.
return &http.Client{
	Transport: &Transport{
		Base:   &http2.Transport{...},
		Source: ReuseTokenSource(nil, src),
	},
}
```

The problem is the second one. We just cannot replace the internals of APIs for creating a client with HTTP/2 client because there might be a Central Dogma server which does not use TLS.
So I think we have to provide ways to use the HTTP/1 client as well for non-TLS servers.
I am planning to replace the old `NewClient*` APIs to use HTTP/2 client internally and add new APIs for HTTP/1 non-TLS client.

/cc @linxGnu and @huydx

**Comment by minwoox** [comment]
Found that there's a way that the client can use h2c. So I think I can just replace to use HTTP/2 client. :)

**Comment by linxGnu** [comment]
@minwoox I really love your idea of creating new api (more `New*` methods) for additional HTTP/2 (with/without TLS) via h2c. The library is stable and easy to use: https://godoc.org/golang.org/x/net/http2#Transport

And it's also convenient if we let user pass his created-initialized client:
```
func New(client *http.Client)
```
So with HTTP/2 support like your suggestion, user has various way to initialize central dogma client. He could choose HTTP/1, HTTP/2 or passing their own `http.Client` 👍

**Comment by minwoox** [comment]
> `func New(client *http.Client)`

We definitely need this, I will add it, thank you!

As for the HTTP/1 and HTTP/2, I was going to leave the HTTP/1 client.
But it turns out that Golang supports `h2c` as well, so we can do both cleartext and encrypted communication using HTTP/2.
Because of that, I couldn't find a reason to keep the HTTP/1 client. Central Dogma server supports HTTP/2 and a user uses this client only to communicate with the server. So I want to completely remove it.
Please let me know if we have to keep it or not.

**Comment by linxGnu** [comment]
I agree with you @minwoox 👍 

Beside that, we need to add metric support, maybe via: https://github.com/armon/go-metrics. I will make a PR after new year holiday 🤔

**Comment by trustin** [comment]
OK to support HTTP/2 only IMO, although it'd be nice to have HTTP/1 support.

**Comment by minwoox** [comment]
> I will make a PR after new year holiday 🤔

Great. I'm looking forward to it. 👍

**Comment by trustin** [comment]
Fixed via #20
